script/release/release/images.py

```diff
@@ -42,33 +47,46 @@ def build_images(self, repository, files):
             if 'stream' in chunk:
                 print(chunk['stream'], end='')
 
-        print('Building test image (for UCP e2e)')
+        if self.latest:
+            self.docker_client.tag(
+                '{image}:{tag}'.format(image=compose_image_base_name, tag=self.version),
+                '{image}:{tag}'.format(image=compose_image_base_name, tag='latest')
+            )
+
+        print('Building test image (debian based for UCP e2e)')
+        compose_tests_image_base_name = compose_image_base_name + '-tests'
+        ucp_test_image = '{image}:{tag}'.format(image=compose_tests_image_base_name, tag=self.version)
         logstream = self.docker_client.build(
-            REPO_ROOT, tag='docker-compose-tests:tmp', decode=True
+            REPO_ROOT,
+            tag=ucp_test_image,
+            target='build',
+            buildargs={
+                'BUILD_PLATFORM': 'debian',
+                'GIT_COMMIT': git_sha,
+            },
+            decode=True
         )
         for chunk in logstream:
             if 'error' in chunk:
                 raise ScriptError('Build error: {}'.format(chunk['error']))
             if 'stream' in chunk:
                 print(chunk['stream'], end='')
 
-        container = self.docker_client.create_container(
-            'docker-compose-tests:tmp', entrypoint='tox'
-        )
-        self.docker_client.commit(container, 'docker/compose-tests', 'latest')
         self.docker_client.tag(
-            'docker/compose-tests:latest', 'docker/compose-tests:{}'.format(self.version)
+            '{image}:{tag}'.format(image=compose_tests_image_base_name, tag=self.version),
+            '{image}:{tag}'.format(image=compose_tests_image_base_name, tag='latest')
         )
-        self.docker_client.remove_container(container, force=True)
-        self.docker_client.remove_image('docker-compose-tests:tmp', force=True)
 
     @property
     def image_names(self):
-        return [
+        images = [
             'docker/compose-tests:latest',
             'docker/compose-tests:{}'.format(self.version),
             'docker/compose:{}'.format(self.version)
         ]
+        if self.latest:
+            images.append('docker/compose:latest')
```

Comment:
As part of this, we should add Debian tags for images. Taking the upcoming 1.25.0 release as an example, we should have:
```
latest [Alpine based 1.25.0]
debian [Debian based 1.25.0]
1.25.0, 1.25.0-alpine [Alpine based 1.25.0]
1.25.0-debian [Debian based 1.25.0]
```

I think that's the minimum level of granularity that we can go with for now. If there's demand, we can add more explicit tags like: `1.25.0-alpine3.9.3` or `1.25.0-stretch-20190326-slim`.

I hesitate doing this now as it adds something else that we'll need to maintain so would prefer some user demand first.

WDYT @thaJeztah?

Reply:
Makes sense; perhaps we should follow the official images tagging scheme (taking  https://hub.docker.com/_/golang/ as example);

Alpine based:
 - `1.25.0`, `1.25`, `1`, `latest`
 - `1.25.0-alpine3.9`, `1.25-alpine3.9`, `1-alpine3.9`, `alpine3.9`
 - `1.25.0-alpine`, `1.25-alpine`, `1-alpine`, `alpine`

Debian based:
- `1.25.0-stretch`, `1.25-stretch`, `1-stretch`, `stretch`


Note:

- Golang has the "default" reversed (`latest` refers to Debian based); we might have to decide if we want to follow that pattern, or be explicit about `alpine` being the preferred
- Will there be Windows based variants? In that case, using `latest` (and tags without `-distroX`) should likely refer to Windowsservercore


Calling @tianon for advice 🤗

Reply:
FWIW, looking at the Golang repo; I wondered why `stretch` was picked, and not `debian-stretch` (so `debian-stretch` -> `debian` (more specific -> less specific), which would be in line with `alpine3.9` -> `alpine` (more specific -> less specific))

Reply:
What I just implemented is...
Given `1.25.0` as an example version, the following tags are produced:
For Alpine: `1.25.0-alpine`, `1.25.0`, `alpine`, `latest`
For Debian: `1.25.0-debian`, `debian`

Knowing that the following tags will roll when releasing the new version:
`alpine`, `latest`, `debian`

I suggest implementing the intermediate tags (like `1-alpine` or `1.25-debian`) in another PR, since it looks more like a new feature than something related to this one.

Reply:
> FWIW, looking at the Golang repo; I wondered why `stretch` was picked, and not `debian-stretch`, so
> 
> * `debian-stretch` -> `debian` (more specific -> less specific), which would be in line with;
> * `alpine3.9` -> `alpine` (more specific -> less specific)

So, there's some history here.  The `stretch` alias was added because we needed a way to help users transition off Jessie, so we tagged with suite and had both for a bit.  Then Alpine had the same issue, so we added similar Alpine tags.  For repositories where the suite isn't as important (like `redis`), I believe we went with simply `debian` (my top-of-the-head example here might be wrong, I'm writing this from my phone, but there's at least one that does that).

The reason Debian is typically the default is because many upstreams don't officially support Alpine and/or musl (Go and Python being very notable examples; Go has upstream tests that fail hard on Alpine with nobody working on fixing them and Python doesn't seem to care much that Alpine even exists, so functionality breaks or acts funny in some cases).

So, all that is really to say I'd recommend proceeding with an Alpine default with extreme caution for something based on Python unless there's a really comprehensive integration test to verify the result. 👍

Reply:
Thanks Tianon!